controller/src/main/java/io/pravega/controller/server/rest/ModelHelper.java

```diff
@@ -57,8 +59,13 @@ public static final StreamConfiguration getCreateStreamConfig(final CreateStream
                             createStreamRequest.getRetentionPolicy().getValue() * 1024 * 1024);
                     break;
                 case LIMITED_DAYS:
-                    retentionPolicy = RetentionPolicy.byTime(
-                            Duration.ofDays(createStreamRequest.getRetentionPolicy().getValue()));
+                    long retentionInDays = createStreamRequest.getRetentionPolicy().getValue();
+                    Duration retentionDuration = (retentionInDays == 0) ?
+                            Duration.ofDays(createStreamRequest.getRetentionPolicy().getTimeBasedRetention().getDays())
```

Comment:
this can get NPE if getTimeBasedRetention is null.. do a precondition check before accessing its fields

Reply:
For old requests timeBasedRetention could be null..... so instead of a precondition check I have  added an explicit null check.